### ISSUE TITLE
When type-checking fails on code-gen. Location is lost, even filaname.

### ISSUE THREAD

**Issue by davesnx** [davesnx/styled-ppx#294]
Getting one of those errors is completely useless. 

```jsx
[re] FAILED: source/components/Greet-TryStyledPpx.cmj source/components/Greet-TryStyledPpx.cmi /Users/davesnx/Code/github/try-styled-ppx/source/components/Greet.bs.js
[re] /Users/davesnx/Code/github/try-styled-ppx/node_modules/bs-platform/darwin/bsc.exe -bs-package-name try-styled-ppx -bs-ns TryStyledPpx  -bs-package-output commonjs:source/components -color always -bs-suffix -I . -I source/components -I source/pages -I source -I /Users/davesnx/Code/github/try-styled-ppx/node_modules/reason-react/lib/ocaml -I /Users/davesnx/Code/github/try-styled-ppx/node_modules/bs-css-emotion/lib/ocaml -I /Users/davesnx/Code/github/try-styled-ppx/node_modules/bs-css/lib/ocaml -warn-error +5 -bs-super-errors -bs-no-version-header -o source/components/Greet-TryStyledPpx.cmj source/components/Greet.reast
[re]
[re]   We've found a bug for you!
[re]   (No file name)
[re]
[re]   This has type:
[re]     CssJs.rule (defined as Css_Js_Core.rule)
[re]   But somewhere wanted:
[re]     string
[re]
[re] FAILED: subcommand failed.
[re] >>>> Finish compiling(exit: 1)
```

**Comment by davesnx** [comment]
I already tried to fix this in two ways, and both weren't not successful.

1. Needed to set the loc before metaquote. Wrapped al declaration_to_emotion into a Functor: https://gist.github.com/davesnx/0952ce03fabfe7acf7f3ff53df565676. Getting a _none_ error after the refactor.
2. Tried to pass location from css_to_emotion into a single property and got same results.

I'm thinking if there's some issue in ppxlib.
